insights/client/connection.py

```diff
@@ -692,7 +692,8 @@ def upload_archive(self, data_collected, duration, cluster=None):
 
         logger.debug("Upload status: %s %s %s",
                      upload.status_code, upload.reason, upload.text)
-        the_json = json.loads(upload.text)
+        if upload.status_code == 201:
+           the_json = json.loads(upload.text)
         try:
```

Comment:
This try/except block should go under the new `if` block as well.  We should probably better handle the case where the `status_code` is not 201.  Not sure if that is done in the calling method or not, but we should somehow indicate that the upload has failed.